src/main/java/com/google/api/codegen/metacode/InitCodeNode.java

```diff
@@ -157,12 +157,39 @@ public static InitCodeNode createSingletonList(String key) {
     return InitCodeNode.createWithChildren(key, InitCodeLineType.ListInitLine, child);
   }
 
+  public static InitCodeNode newRoot() {
+    return new InitCodeNode("root", InitCodeLineType.StructureInitLine, InitValueConfig.create());
+  }
+
   /*
    * Constructs a tree of objects to be initialized using the provided context, and returns the root
    */
   public static InitCodeNode createTree(InitCodeContext context) {
-    List<InitCodeNode> subTrees = buildSubTrees(context);
-    InitCodeNode root = createWithChildren("root", InitCodeLineType.StructureInitLine, subTrees);
+    Preconditions.checkArgument(
+        context.initFields() != null || context.outputType() != InitCodeOutputType.FieldList,
+        "init field array is not set for flattened method");
+    InitCodeNode root = newRoot();
+
+    if (context.initFields() != null) {
+      for (FieldModel field : context.initFields()) {
+        String nameString = field.getNameAsParameter();
+        InitValueConfig initValueConfig = context.initValueConfigMap().get(nameString);
+        if (initValueConfig == null) {
+          root.mergeChild(InitCodeNode.createWithName(nameString, nameString));
```

Comment:
It will happen recursively, but the new node doesn't have children, so it won't actually recur.

We still need `mergeChild` for `additionalInitCodeNodes` which might actually be recursive though.